sdv/multi_table/base.py

```diff
@@ -82,3 +83,74 @@ def update_table_parameters(self, table_name, table_parameters):
     def get_metadata(self):
         """Return the ``MultiTableMetadata`` for this synthesizer."""
         return self.metadata
+
+    def _validate_foreign_keys(self, data):
+        errors = []
+        for relation in self.metadata._relationships:
+            child_table = data[relation['child_table_name']]
+            parent_table = data[relation['parent_table_name']]
+            child_row = child_table[relation['child_foreign_key']]
+            parent_row = parent_table[relation['parent_primary_key']]
+            missing_values = child_row[~child_row.isin(parent_row)].unique()
+            if any(missing_values):
+                message = ', '.join(missing_values[:5].astype(str))
+                if len(missing_values) > 5:
+                    message = f'({message}, + more).'
+                else:
+                    message = f'({message}).'
+
+                errors.append(
+                    f"Error: foreign key column '{relation['child_foreign_key']}' contains "
+                    f'unknown references: {message}'
+                )
+
+        return errors
+
+    def validate(self, data):
+        """Validate data.
+
+        Args:
+            data (dict):
+                A dictionary with key as table name and ``pandas.DataFrame`` as value to validate.
+
+        Raises:
+            ValueError:
+                Raised when data is not of type pd.DataFrame.
+            InvalidDataError:
+                Raised if:
+                    * foreign key does not belong to a primay key
+                    * data columns don't match metadata
+                    * keys have missing values
+                    * primary or alternate keys are not unique
+                    * context columns vary for a sequence key
+                    * values of a column don't satisfy their sdtype
+        """
+        errors = []
+        missing_tables = set(self.metadata._tables) - set(data)
+        if missing_tables:
+            if len(missing_tables) > 1:
+                raise InvalidDataError([
+                    f'The provided data is missing the tables {missing_tables}.'
+                ])
+            else:
+                raise InvalidDataError([
+                    f'The provided data is missing the table {missing_tables}.'])
+
+        for table_name, table_data in data.items():
+            try:
+                self._table_synthesizers[table_name].validate(table_data)
+
+            except (InvalidDataError, ValueError) as error:
+                error_msg = str(error)
+                error_msg = error_msg.replace(
+                    'The provided data does not match the metadata:',
+                    f"The provided data for table '{table_name}' does not match the metadata:"
+                )
+                errors.append(error_msg)
+
+        if errors:
```

Comment:
should we raise these before doing the foreign keys check, or just raise them all together at the bottom

Reply:
@npatki I think the desire is to have all errors returned at once right?

Reply:
The outcome is to throw a single `InvalidDataError` whose message contains all the problems that we can possibly fine.

So this means we should throw it after doing _all_ the checks (including any primary/foreign key checks.). This is similar to what we are doing in `metadata.validate()`.